Unable to create materialized view in Stargate 1.0.28

cqlsh:
```
cassandra@cqlsh:t1> CREATE TABLE aaaa (x int, y text, primary key (x))  with comment = 'cccc1';
cassandra@cqlsh:t1> CREATE MATERIALIZED VIEW mmmm AS SELECT * from aaaa where y is not null PRIMARY KEY ( y, x) ;
NoHostAvailable: 
```

Stargate log:
```
ERROR [Native-Transport-Requests-1] 2021-06-28 13:33:53,863 ErrorMessage.java:384 - Unexpected exception during request
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cannot build MaterializedView, some of required attributes are not set [comment]
	at org.apache.cassandra.utils.FBUtilities.waitOnFuture(FBUtilities.java:442)
	at org.apache.cassandra.service.MigrationManager.announce(MigrationManager.java:572)
	at org.apache.cassandra.service.MigrationManager.announceNewView(MigrationManager.java:383)
	at org.apache.cassandra.cql3.statements.CreateViewStatement.announceMigration(CreateViewStatement.java:336)
	at org.apache.cassandra.cql3.statements.SchemaAlteringStatement.execute(SchemaAlteringStatement.java:123)
	at org.apache.cassandra.cql3.QueryProcessor.processStatement(QueryProcessor.java:225)
	at io.stargate.db.cassandra.impl.StargateQueryHandler.process(StargateQueryHandler.java:160)
	at org.apache.cassandra.transport.messages.QueryMessage.execute(QueryMessage.java:116)
	at io.stargate.db.cassandra.impl.CassandraPersistence$CassandraConnection.lambda$executeRequestOnExecutor$1(CassandraPersistence.java:466)
	at io.stargate.db.cassandra.impl.CassandraPersistence.lambda$runOnExecutor$1(CassandraPersistence.java:278)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$FutureTask.run(AbstractLocalAwareExecutorService.java:162)
	at org.apache.cassandra.concurrent.SEPWorker.run(SEPWorker.java:113)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Cannot build MaterializedView, some of required attributes are not set [comment]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at org.apache.cassandra.utils.FBUtilities.waitOnFuture(FBUtilities.java:438)
	... 13 common frames omitted
Caused by: java.lang.IllegalStateException: Cannot build MaterializedView, some of required attributes are not set [comment]
	at io.stargate.db.schema.ImmutableMaterializedView$Builder.build(ImmutableMaterializedView.java:695)
	at io.stargate.db.schema.MaterializedView.create(MaterializedView.java:35)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.convertMVIndex(AbstractCassandraSchemaConverter.java:265)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.lambda$convertMVIndexes$5(AbstractCassandraSchemaConverter.java:259)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.StreamSpliterators$WrappingSpliterator.forEachRemaining(StreamSpliterators.java:312)
	at java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Streams.java:743)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.convertTable(AbstractCassandraSchemaConverter.java:154)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.lambda$convertTables$0(AbstractCassandraSchemaConverter.java:142)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.convertKeyspace(AbstractCassandraSchemaConverter.java:133)
	at com.google.common.collect.Iterators$8.transform(Iterators.java:799)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at io.stargate.db.schema.ImmutableSchema$Builder.addAllKeyspaces(ImmutableSchema.java:238)
	at io.stargate.db.schema.Schema.create(Schema.java:51)
	at io.stargate.db.datastore.common.AbstractCassandraSchemaConverter.convertCassandraSchema(AbstractCassandraSchemaConverter.java:124)
	at io.stargate.db.datastore.common.AbstractCassandraPersistence.computeCurrentSchema(AbstractCassandraPersistence.java:113)
	at io.stargate.db.datastore.common.AbstractCassandraPersistence.lambda$initialize$0(AbstractCassandraPersistence.java:109)
	at io.stargate.db.cassandra.impl.CassandraPersistence$1.onSchemaChange(CassandraPersistence.java:174)
	at io.stargate.db.cassandra.impl.SimpleCallbackMigrationListener.onCreateView(SimpleCallbackMigrationListener.java:29)
	at org.apache.cassandra.service.MigrationManager.notifyCreateView(MigrationManager.java:225)
	at org.apache.cassandra.config.Schema.addView(Schema.java:735)
	at java.util.HashMap$Values.forEach(HashMap.java:981)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1080)
	at org.apache.cassandra.schema.SchemaKeyspace.updateKeyspace(SchemaKeyspace.java:1470)
	at org.apache.cassandra.schema.SchemaKeyspace.mergeSchema(SchemaKeyspace.java:1434)
	at org.apache.cassandra.schema.SchemaKeyspace.mergeSchema(SchemaKeyspace.java:1403)
	at org.apache.cassandra.schema.SchemaKeyspace.mergeSchemaAndAnnounceVersion(SchemaKeyspace.java:1380)
	at org.apache.cassandra.service.MigrationManager$1.runMayThrow(MigrationManager.java:590)
	at org.apache.cassandra.utils.WrappedRunnable.run(WrappedRunnable.java:28)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.cassandra.concurrent.NamedThreadFactory.lambda$threadLocalDeallocator$0(NamedThreadFactory.java:84)
	... 1 common frames omitted
```

Affects: v1.0.28 (related to: #1009)

--- a/persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java
+++ b/persistence-api/src/main/java/io/stargate/db/query/builder/QueryBuilderImpl.java
@@ -83,7 +83,7 @@ import org.javatuples.Pair;
       @SubExpr(
           name = "index",
           definedAs =
-              "(drop ((materializedView|index) ifExists?)) | (create ((materializedView ifNotExists? asSelect (column+) from)"
+              "(drop ((materializedView|index) ifExists?)) | (create ((materializedView ifNotExists? asSelect (column+) from withComment?)"
                   + " | (index ifNotExists? on column (indexKeys|indexValues|indexEntries|indexFull|indexingType)? (custom options?)?)))"),
     })
 public class QueryBuilderImpl {

--- a/persistence-api/src/main/java/io/stargate/db/schema/MaterializedView.java
+++ b/persistence-api/src/main/java/io/stargate/db/schema/MaterializedView.java
@@ -23,6 +23,11 @@ public abstract class MaterializedView extends AbstractTable implements Index {
   private static final long serialVersionUID = -2999120284516448661L;
 
   public static MaterializedView create(String keyspace, String name, Iterable<Column> columns) {
+    return create(keyspace, name, columns, "");
+  }
+
+  public static MaterializedView create(
+      String keyspace, String name, Iterable<Column> columns, String comment) {
     columns.forEach(
         c -> {
           Preconditions.checkState(
@@ -32,6 +37,7 @@ public abstract class MaterializedView extends AbstractTable implements Index {
         .keyspace(keyspace)
         .name(name)
         .addAllColumns(columns)
+        .comment(comment)
         .build();
   }
 

--- a/persistence-cassandra-4.0/src/test/java/io/stargate/db/cassandra/impl/CassandraPersistenceIT.java
+++ b/persistence-cassandra-4.0/src/test/java/io/stargate/db/cassandra/impl/CassandraPersistenceIT.java
@@ -9,6 +9,7 @@ import java.io.File;
 import java.io.IOException;
 import java.net.URL;
 import java.nio.file.Files;
+import org.apache.cassandra.config.Config;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 
@@ -50,7 +51,9 @@ class CassandraPersistenceIT extends PersistenceTest {
     }
 
     persistence = new CassandraPersistence();
-    persistence.initialize(makeConfig(baseDir));
+    Config config = makeConfig(baseDir);
+    config.enable_materialized_views = true;
+    persistence.initialize(config);
   }
 
   @AfterAll

--- a/persistence-common/src/main/java/io/stargate/db/datastore/common/AbstractCassandraSchemaConverter.java
+++ b/persistence-common/src/main/java/io/stargate/db/datastore/common/AbstractCassandraSchemaConverter.java
@@ -262,7 +262,8 @@ public abstract class AbstractCassandraSchemaConverter<K, T, C, U, I, V> {
   private Index convertMVIndex(String keyspaceName, V view) {
     T table = asTable(view);
     List<Column> columns = convertColumns(keyspaceName, table).collect(Collectors.toList());
-    return MaterializedView.create(keyspaceName, tableName(table), columns);
+    String comment = comment(table);
+    return MaterializedView.create(keyspaceName, tableName(table), columns, comment);
   }
 
   private Stream<UserDefinedType> convertUserTypes(String keyspaceName, Iterable<U> userTypes) {

--- a/persistence-test/src/main/java/io/stargate/it/PersistenceTest.java
+++ b/persistence-test/src/main/java/io/stargate/it/PersistenceTest.java
@@ -80,6 +80,7 @@ import io.stargate.db.schema.Column;
 import io.stargate.db.schema.ImmutableTupleType;
 import io.stargate.db.schema.ImmutableUserDefinedType;
 import io.stargate.db.schema.Keyspace;
+import io.stargate.db.schema.MaterializedView;
 import io.stargate.db.schema.ParameterizedType;
 import io.stargate.db.schema.Schema;
 import io.stargate.db.schema.Table;
@@ -663,6 +664,43 @@ public abstract class PersistenceTest {
     assertThat(actualTable.indexes().size()).isEqualTo(expectedTable.indexes().size());
     assertThat(actualTable.index("byD").toString())
         .isEqualTo(expectedTable.index("byD").toString());
+  }
+
+  @Test
+  public void testMaterializedView() {
+    createKeyspace();
+    dataStore
+        .queryBuilder()
+        .create()
+        .table(keyspace, table)
+        .column("a", Int, PartitionKey)
+        .column("b", Varchar, Clustering)
+        .withComment("test-comment1")
+        .build()
+        .execute()
+        .join();
+    dataStore
+        .queryBuilder()
+        .create()
+        .materializedView(keyspace, "test_MV")
+        .asSelect()
+        .column("b", PartitionKey)
+        .column("a", Clustering)
+        .from(keyspace, table)
+        .withComment("test-comment2")
+        .build()
+        .execute()
+        .join();
+
+    MaterializedView mv = dataStore.schema().keyspace(keyspace).materializedView("test_MV");
+    assertThat(mv).isNotNull();
+    assertThat(mv.name()).isEqualTo("test_MV");
+    assertThat(mv.comment()).isEqualTo("test-comment2");
+    assertThat(mv.columns()).hasSize(2);
+    assertThat(mv.column("b").kind()).isEqualTo(PartitionKey);
+    assertThat(mv.column("b").type()).isEqualTo(Varchar);
+    assertThat(mv.column("a").kind()).isEqualTo(Clustering);
+    assertThat(mv.column("a").type()).isEqualTo(Int);
   }
 
   @Disabled("Disabling for now since it currently just hangs")

